### PR TITLE
feat: solidity.vim of syntax, indent, ftplugin

### PR DESCRIPTION
**Pull request by coti-z**
hi, I looked it up and found that the syntax solidity file was missing. How about supporting solidity syntax?

### CODE REVIEW

**Comment by brammool** [comment]
Is this one better than the previous PR?

**Comment by brammool** [comment]
The file does not set b:current_syntax

**Comment by cothi** [comment]
There was a bit of confusion about licensing. So I made a mistake, then I PR it again.
 I corrected `the source code typo`, and  I added a modified source code
, so please check it out

thank you

**Comment by k-takata** [comment]
The original repository (https://github.com/tomlion/vim-solidity) has not only the syntax file but also the indent file and the ftplugin file.
Are you going to take over the maintenance of the syntax file only?

**Comment by cothi** [comment]
I added the necessary file for solidity (`indent`, `ftplugin`)

File: runtime/ftplugin/solidity.vim [hunk 1]
```diff
@@ -0,0 +1,7 @@
+" Vim filetype plugin file
+" Language: Solidity
+" Previous Maintainer: Cothi (jiungdev@gmail.com) 
+" Original author: tomlion (https://github.com/tomlion/vim-solidity)
+" Last Changed: 2022 Sep 27
+
+setlocal commentstring=//\ %s
```
Comment (dkearns) [inline runtime/ftplugin/solidity.vim:7]: `b:undo_ftplugin` needs to be set. See `:help undo_ftplugin`

Setting `'comments'` as well while you're here would be a good idea.

File: runtime/ftplugin/solidity.vim [hunk 2]
```diff
@@ -0,0 +1,7 @@
+" Vim filetype plugin file
+" Language: Solidity
+" Previous Maintainer: Cothi (jiungdev@gmail.com) 
+" Original author: tomlion (https://github.com/tomlion/vim-solidity)
```
Comment (dkearns) [inline runtime/ftplugin/solidity.vim:4]: Simply using `Author:` here is more 'standard' and is recognised by the syntax highlighters.

File: runtime/indent/solidity.vim [hunk 3]
```diff
@@ -1,9 +1,9 @@
 " Vim indent file
-" Language: 		Solidity
-" Acknowledgement: 	Based off of vim-javascript
-" Maintainer: 		Cothi (jiungdev@gmail.com)
-" Original Author: 	tomlion (https://github.com/tomlion/vim-solidity)
-" Last Changed: 	2022 Sep 27
+" Language: Solidity
+" Acknowledgement: Based off of vim-javascript
+" Maintainer: Cothi (jiungdev@gmail.com) 
+" Original author: tomlion (https://github.com/tomlion/vim-solidity)
```
Comment (dkearns) [inline runtime/indent/solidity.vim:5]: Simply using Author: here is more 'standard' and is recognised by the syntax highlighters.

**Review by dkearns — CHANGES_REQUESTED** [review]
The indent plugin needs to have `b:undo_indent` set as well to undo the setting of `'smartindent'`, `'indentexpr'` and `'indentkeys'`.  See `:help undo_indent` for details.

Could you please also add yourself to https://github.com/vim/vim/blob/master/.github/CODEOWNERS?

**Comment by chrisbra** [comment]
@cothi any updates please?

**Pull request closed without merging**